expr/functions.go

```diff
@@ -0,0 +1,63 @@
+package expr
+
+import (
+	"errors"
+	"fmt"
+	"math"
+
+	"github.com/brimsec/zq/zng"
+)
+
+type Function func([]NativeValue) (NativeValue, error)
+
+var ErrWrongArgc = errors.New("wrong number of arguments")
+var ErrBadArgument = errors.New("bad argument")
+
+var allFns = []struct {
+	name string
+	fn   Function
+}{
+	{"Math.sqrt", mathSqrt},
+}
+
+var allFnsMap map[string]Function
+var fnsInited = false
+
+func lookupFunction(name string) *Function {
+	if !fnsInited {
+		allFnsMap = make(map[string]Function)
+		for _, f := range allFns {
+			allFnsMap[f.name] = f.fn
+		}
+	}
+	fn, ok := allFnsMap[name]
+	if ok {
+		return &fn
+	}
+	return nil
+}
+
+func mathSqrt(args []NativeValue) (NativeValue, error) {
+	if len(args) < 1 || len(args) > 1 {
+		return NativeValue{}, fmt.Errorf("Math.sqrt: %w", ErrWrongArgc)
+	}
+
+	var x float64
+	switch args[0].typ.ID() {
+	case zng.IdFloat64:
+		x = args[0].value.(float64)
+	case zng.IdInt16, zng.IdInt32, zng.IdInt64:
+		x = float64(args[0].value.(int64))
+	case zng.IdByte, zng.IdUint16, zng.IdUint32, zng.IdUint64:
+		x = float64(args[0].value.(uint64))
+	default:
+		return NativeValue{}, fmt.Errorf("Math.sqrt: %w", ErrBadArgument)
+	}
+
+	r := math.Sqrt(x)
+	if math.IsNaN(r) {
```

Comment:
for the other reviewers: I asked @aswan about why we wanted to return errors for NaN's. That's not something we need to do forever, but we're not ready yet to declare NaN-like values in the language. So I think he'll add a comment here, and we'll file an issue to consider how we'll handle NaN-like values.